### PR TITLE
chore(test): migrate `watch-native-attributes` to wdio

### PR DESCRIPTION
**Pull request by tanner-reits**
## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

Migratest `test/karma/test-app/watch-native-attributes` to wdio

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: test/wdio/watch-native-attributes/cmp.test.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  it('triggers the callback for the watched attribute', async () => {
+    await $('watch-native-attributes').waitForExist();
+
+    const cmp = document.querySelector('watch-native-attributes');
+    expect(cmp.innerText).toBe('Label: myStartingLabel\n\nCallback triggered: false');
+
+    cmp.setAttribute('aria-label', 'myNewLabel');
+    await browser.pause(100);
+
+    expect(cmp.innerText).toBe('Label: myNewLabel\n\nCallback triggered: true');
```
Comment (christian-bromann) [inline test/wdio/watch-native-attributes/cmp.test.tsx:20]: I would suggest something like this to leverage the async matchers.
```suggestion
    const $cmp = $('watch-native-attributes')
    await $cmp.waitForExist();

    await expect($cmp).toHaveText('Label: myStartingLabel\n\nCallback triggered: false');

    const cmp = document.querySelector('watch-native-attributes');
    cmp.setAttribute('aria-label', 'myNewLabel');

    await expect(cmp).toHaveText('Label: myNewLabel\n\nCallback triggered: true');
```

I looking into ways to easier retrieve the raw DOM element from the `WebdriverIO.Element` object to allow us to just define the element once.

**Review by christian-bromann — COMMENTED** [review]
One suggestion

File: test/wdio/watch-native-attributes/cmp.test.tsx (hunk 1 above)
Comment (tanner-reits) [reply test/wdio/watch-native-attributes/cmp.test.tsx:20]: Is the `$` variable prefix the recommended convention when using async queries? Just curious!

**Review by alicewriteswrongs — APPROVED** [review]
lgtm!

File: test/wdio/watch-native-attributes/cmp.test.tsx (hunk 1 above)
Comment (christian-bromann) [reply test/wdio/watch-native-attributes/cmp.test.tsx]: not really .. but in the context of using `HTMLElement` and `WebdriverIO.Element` objects it seems like a good idea to do but not a convention.

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**